### DISCUSSION TITLE
Minimal GDAL deployment to Windows Server 2012 R2 development server is not working, causing an APPCRASH event

### DISCUSSION THREAD

**Discussion by trippetoe** [MaxRev-Dev/gdal.netcore#173] [Q&A] [answered]
I have .net core 8 web api project where I have installed gdal via nuget:
```
 <ItemGroup>
    <PackageReference Include="MaxRev.Gdal.Core" Version="3.10.0.306" />
    <PackageReference Include="MaxRev.Gdal.WindowsRuntime.Minimal" Version="3.10.0.306" />
```
	
The app is doing some simple coordinate system transformations, and it works as expected when running on my local machine.

However, things go bad when i try to deploy to my development server (64-bit Windows Server 2012 R2). To build the project I'm using the command: `dotnet publish --configuration Development --output ./publish -p:AspNetCoreEnvironment=Development`.  After I copy/paste the `publish` folder to the development server, and I try to view the swagger home page, the app trys to start, but crashes at the call to 

```
GdalBase.ConfigureAll();
```
 Two errors are recorded in Event Viewer, and the App Pool shut downs.

**Application Error 1**:
```
Windows cannot access the file  for one of the following reasons: there is a problem with the network connection, the disk that the file is stored on, or the storage drivers installed on this computer; or the disk is missing. Windows closed the program IIS Worker Process because of this error.

Program: IIS Worker Process
...

```
**Application Error 2**:
```
Faulting application name: w3wp.exe, version: 8.5.9600.16384, time stamp: 0x5215df96
Faulting module name: gdal.dll, version: 3.10.0.0, time stamp: 0x6735118c
Exception code: 0xc000001d
Fault offset: 0x000000000004eb3d
Faulting process id: 0xf88
Faulting application start time: 0x01db3a052a92f4a3
Faulting application path: c:\windows\system32\inetsrv\w3wp.exe
Faulting module path: C:\inetpub\wwwroot\<appname>\runtimes\win-x64\native\gdal.dll
Report Id: 68e5297a-a5f8-11ef-8118-005056af7681
Faulting package full name: 
Faulting package-relative application ID: 
``` 

I've tried a couple things that haven't had any discernable impact
- I verified that `...\runtimes\win-x64\native\gdal.dll` exists
- Granted read/execute permissions for the AppPool identity to the application folder. I verified that those permissions extend to gdal.dll
- installed Microsoft Visual C++ Redistributable (a recommendation from my work mate ChatGPT)

Any thoughts on what is going wrong either on the server or with my publish process?

**Comment by MaxRev-Dev** [comment]
What I would recommend is to try to install GitBash and run ldd (as described [here](https://github.com/MaxRev-Dev/gdal.netcore/issues/143#issuecomment-2286585656)). You need to run it on 'gdal.dll'.

I can confirm that it works on Windows Server 2022. However, if you're **running a VM** - you need to enable [AVX/AVX2 processor extensions](https://github.com/MaxRev-Dev/gdal.netcore/issues/150#issuecomment-2452828200). Or your target machine should support AVX.

Sometimes it requires to cleanup the project and publish it again - ref: https://github.com/MaxRev-Dev/gdal.netcore/issues/144

There were some issues with VC runtime in previous versions, but the latest one contains them, so it shouldn't require explicit installation and will use the binaries in app working directory.

**Reply by trippetoe** [reply]
@MaxRev-Dev Thank you! for your super fast reply. I'm trying the `ldd` command. It's not a command i was familiar with. Currently its been "running" for about 30 mins or so with no signs of progress. How will i know when its done?

![image](https://github.com/user-attachments/assets/4a490ef2-be10-4a17-bf39-3959c9cd62a1)

In the meantime i tried deleting, cleaning, and rebuilding the project with the same over-all results - it works fine on my laptop and crashes on the server. 

I'll get back to you if anything exciting happens with the `ldd` command.

**Reply by MaxRev-Dev** [reply]
No need, you can stop it. it usually takes less than a second to complete.
It looks like there is a dependency collision somewhere causing an infinite loop in ldd

**Reply by MaxRev-Dev** [reply, marked as answer]
@trippetoe As windows server 2012 is **already EOL** and not supported I don't see any obvious reasons to investigate it.

Why it doesn't work? I suppose it's missing VCRT v140 version which is shipped with packages. But some files are still coming from system (like ntdll). There can be missing methods and incompatible ABI. And those redistributables [are binary compatible](https://learn.microsoft.com/en-us/cpp/porting/binary-compat-2015-2017?view=msvc-170&viewFallbackFrom=vs-2019) with VCRT 2015+.

You have two options:
- Move to the latest windows server (at least 2016).
- Use preinstalled GDAL binaries (no need for WindowsRuntime package then). You can download them from GisInternals as well.  [Read this thread](https://github.com/MaxRev-Dev/gdal.netcore/issues/156#issuecomment-2429316745).

**Reply by trippetoe** [reply]
@MaxRev-Dev - Totally understand about not wanting to sink energy or time into supporting EOL software. 

I appreciate your willingness to throw out a couple of ideas. And really do appreciate you and the work you do making GDAL available for .net developers.
